### DISCUSSION TITLE
Tasmota ESP32 ADC with lookup table

### DISCUSSION THREAD

**Discussion by boozeman** [arendst/Tasmota#13822] [Q&A] [answered]
Hi there,

TL;DR;

Is it possible to use special LUT-table for ESP32 Analog Input described at here: [https://github.com/e-tinkers/esp32-adc-calibrate](url)

Background:

I have one analog MP3v5010DP Pressure Sensor in use and with special made LUT-table for ESP32 ADC,  the readings are some what linear.  Sensor measures 150 litre water tank pressure with adjustable tank empty and tank full readings with MQTT. Mapfloat function converts readins for litres. I know when to fill up Greenhouse water tank at summer and it is good to get some statistics in Home Assistant too.  :)

My main problem is that i do not managed to find any working ESP32 + PubSubClient + TLS examples so far. 

I managed to get TLS work with Tasmota with pretty good [Self-signed-Mosquitto](https://tasmota.github.io/docs/Self-signed-Mosquitto/) instructions and build custom firmware.  Greenhouse special ESP32 includes some BME280 and SHT31 sensors, so the ease of Tasmota attracted to me for this special use too..

Cheers

**Comment by s-hadinger** [comment, marked as answer]
You can certainly do this in Berry scripting. It should not be too complex to do. However this will consume a significant amount of RAM. It should be OK if you don't have other ram consuming features (display...). No problem if you of PSRAM

**Reply by Jason2866** [reply]
I doubt the LUT makes the ADC results much better, since the the calibration is done with DAC from the ESP32. The DAC from the ESP32 isnt great either. Much effort to get a doubtable result.
Way easier and for sure more precise. Use a extern ADC connected via i2c.

**Reply by boozeman** [reply]
The LUT-tables I was used last summer was ESP32 specific and the pressure curve was always nice and linear. I tested also with ADS1115 but the hysteresis was somewhat awfull. (changes per measure are about 2-10l) No matter what settings I was using with that chip. I will test that again and figure out what was wrong in last summer.

Below last 150 litres before the growing season was stopped.

![image](https://user-images.githubusercontent.com/40159019/143684147-7dd7edb7-4b5c-487d-9efb-20598bc65b7b.png)
